README.md

```diff
@@ -1 +1,189 @@
+>
+> — [DAG-JSON Spec](https://ipld.io/specs/codecs/dag-json/spec/)
+
+As it is critical for signatures guard against various attacks, the assumptions around canonical encoding MUST be enforced.
+
+## 2.1 Signing CIDs
+
+Rather than validating the inline IPLD, replacing the data with a CID link to the content MAY be used instead. Note while this is very safe (as it impractical to alter a signed hash), this approach mixes data layout with security, and may have a performance, disk, and networking impacts.
+
+## 2.2 Raw (Noncanonicalized) Data
+
+Canonicalization is not required if data is encoded as raw bytes (multicodec `0x00`). The exact bytes are already present, and MUST not be changed.
+
+# 3 Varsig Format
+
+A varsig is a bytestring that includes the following information:
+
+| Segment Name              | Type     | Description                                     | Required |
+|---------------------------|----------|-------------------------------------------------|----------|
+| Varsig Prefix             | `0xD000` | The multicodec varsig prefix                    | Yes      |
+| Public Key Prefix         | `Varint` | The multicodec prefix for the public key type   | Yes      |
+| Hash Prefix               | `Varint` | The multicodec prefix for the hash algorithm    | Yes      |
+| Hash Length               | `Varint` | The hash length                                 | Yes      |
+| Content Multicodec Prefix | `Varint` | The IPLD encoding uses to canonicalize the data | Yes      |
+| Raw Signature             | `Varint` | The raw signature                               | Yes      |
+  
+## 3.1 Segments
+
+### 3.1.1 Varsig Prefix
+
+The varsig prefix MUST be `0xD000`.
+
+### 3.1.2 Public Key Prefix
+
+The [multicodec](https://github.com/multiformats/multicodec) prefix for the public key type associated with the signature.
+
+For example: `0xE7` for secp256k1, or `0x1205` for RSA.
+
+### 3.1.3 Hash Prefix
+
+The [multicodec](https://github.com/multiformats/multicodec) hash prefix used by the signature. This is the first segment of a [multihash](https://github.com/multiformats/multihash).
+
+For example: `0x16` for SHA3-256, or `0x1E` for BLAKE3
+
+### 3.1.4 Hash Length
+
+The hash length used by the signature. This is the second segment of a [multihash](https://github.com/multiformats/multihash).
+
+### 3.1.5 Content Multicodec Prefix
+
+The [multicodec](https://github.com/multiformats/multicodec) prefix of the encoding scheme used.
+
+For example: `0x00` for raw bytes (no special encoding), `0x0129` for DAG-JSON, and `0x70` for DAG-PB.
+
+### 3.1.6 Raw Signature
+
+The raw signature bytes
+
+## 3.2 Byte Segments
+
+```xml
+<varint 0xD000><varint multicodec_key_prefix><varint multicodec_hash_prefix><varint multicodec_hash_length><varint multicodec_prefix><varint raw_hash><varint raw_signature>
```

Comment:
`0xD000` is what's currently in the multicodec table, and `0xD0` is taken. This version of varsig doesn't need the various signature types, since we can use the public key types, so maybe we drop them from the table and reassign the varsig prefix to a different one-byte varsig-as-multiformat prefix?